diff -urN qt-all-opensource-src-4.3.5.orig/src/gui/styles/qwindowsvistastyle.cpp qt-all-opensource-src-4.3.5/src/gui/styles/qwindowsvistastyle.cpp
--- qt-all-opensource-src-4.3.5.orig/src/gui/styles/qwindowsvistastyle.cpp	2008-05-26 13:06:56.000000000 +0400
+++ qt-all-opensource-src-4.3.5/src/gui/styles/qwindowsvistastyle.cpp	2008-09-04 18:17:12.000000000 +0400
@@ -1042,7 +1042,9 @@
                 stateId = MBI_HOT;
 
             XPThemeData theme2(widget, painter, QLatin1String("MENU"), MENU_BARITEM, stateId, option->rect);
+            if (!widget->property("hide-menu-popupgutter").toBool()) {
             d->drawBackground(theme2);
+            } // hide-menu-popupgutter
 
             if (!pix.isNull())
                 drawItemPixmap(painter, mbi->rect, alignment, pix);
